src/freenet/node/OpennetManager.java

```diff
@@ -546,6 +589,39 @@ public boolean wantPeer(OpennetPeerNode nodeToAddNow, boolean addAtLRU, boolean
 				return false;
 			}
 		}
+		// Don't connect to peers in excluded countries
+		if (nodeToAddNow != null) {
+			Peer[] handshakeIPs = nodeToAddNow.getHandshakeIPs();
+			if (handshakeIPs != null) {
+				for (Peer peer : handshakeIPs) {
+					if (peer == null) {
+						continue;
+					}
+					FreenetInetAddress address = peer.getFreenetAddress();
+					if (address == null) {
+						continue;
+					}
+					InetAddress inetAddress = address.getAddress(false);
+
+					IPConverter ipConverter = IPConverter.getInstance(NodeFile.IPv4ToCountry.getFile(node));
+					try {
+						Country country = ipConverter.locateIP(inetAddress.getAddress());
+						if (country != null) {
+							if (excludedCountries.contains(country.toString().toUpperCase())) {
+								Logger.error(this, "Not connecting to Opennet peer " + inetAddress.toString() + " in country " + country.toString());
```

Comment:
error() should be used to indicate bugs in the code, not for things which the user told us to do.
I'd say:
- Use normal() if the frequency of this occurring is above "every few seconds"
- If it occurs more frequently than that, use "if(logMINOR) Logger.minor(...)"